src/Common/tests/System/PlatformDetection.cs

```diff
@@ -456,5 +456,28 @@ public bool IsInRange(Version version)
                 return version >= Start && (Finish == null || version <= Finish);
             }
         }
+
+        public static bool IsEnvironmentCommandLineArgsSupported => s_IsEnvironmentCommandLineArgsSupported.Value;
+
+        private static readonly Lazy<bool> s_IsEnvironmentCommandLineArgsSupported = new Lazy<bool>(
+            delegate ()
+            {
+                if (!PlatformDetection.IsNetNative)
+                    return true;
+
+                try
+                {
+                    Environment.GetCommandLineArgs();
+                }
+                catch (PlatformNotSupportedException)
+                {
+                    return false;
+                }
+                catch
+                {
+                    return true; // Hmm - threw something but not PNSE. Let the tests loose on it and report what they find...
```

Comment:
Calling `GetCommandLine` is susceptible to someone writing over the returned buffer. `Windows.h` is very inviting to that sort of changes because the string is returned as `LPWSTR` as opposed to `LPCWSTR`.

I'm fine with using that, as long as the CoreRT implementation stays untouched (where we fetch it as part of program startup, and if we don't own the startup, the host can provide one).